res/controllers/Denon-MC7000-scripts.js

```diff
@@ -888,6 +894,41 @@ MC7000.crossFaderCurve = function(control, value) {
     script.crossfaderCurve(value);
 };
 
+// Update state on deck changes
+MC7000.switchDeck = function(channel, control, value, status, group) {
+    var deckOffset = script.deckFromGroup(group) - 1;
+    var isTopDeck = deckOffset < 2;
+    var side = deckOffset % 2;
+
+    var previousDeckOffset;
+    switch (deckOffset) {
+    case 0:
+        previousDeckOffset = 2;
+        break;
+    case 1:
+        previousDeckOffset = 3;
+        break;
+    case 2:
+        previousDeckOffset = 0;
+        break;
+    case 3:
+        previousDeckOffset = 1;
+        break;
+    }
+
```

Comment:
IMO both suggestions are valid, choose whichever you think conveys intent better. 
```suggestion
    var previousDeckOffset = (deckOffset + 2) % 4;
```
```suggestion
    var previousDeckOffset = deckOffset + (isTopDeck ? 2 : -2);
```